dist/tools/compile_test/compile_test.py

```diff
@@ -100,10 +106,10 @@ def build_all():
                                    env=subprocess_env)
 
                 results, results_with_output = tee(get_results_and_output_from(subprocess.stdout))
-                results = groupby(sorted(results), lambda (outcome, board, output): outcome)
-                results_with_output = filter(lambda (outcome, board, output): output.getvalue(), results_with_output)
-                failed_with_output = filter(lambda (outcome, board, output): 'failed' in outcome, results_with_output)
-                success_with_output = filter(lambda (outcome, board, output): 'success' in outcome, results_with_output)
+                results = groupby(sorted(results), lambda res: res[0])
```

Comment:
I know this is a bit ugly and makes it less readable, but it was the only way I could get it to work in both Python 2.x and Python 3.x without resorting to writing two separate `if` branches.